BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/ble/beacon/Beacon.java

```diff
@@ -137,23 +139,25 @@ public boolean hasBeenSeenSince(long timestamp) {
         return getLatestAdvertisingPacket().getTimestamp() > timestamp;
     }
 
-    public float getMeanRssi() {
-        return getMeanRssi(3, TimeUnit.SECONDS);
+    public float getRssi(RssiFilter filter) {
+        return filter.filter(this);
     }
 
-    public float getMeanRssi(long amount, TimeUnit timeUnit) {
-        List<AdvertisingPacket> recentAdvertisingPackets = (List<AdvertisingPacket>) getAdvertisingPacketsFromLast(amount, timeUnit);
-        int[] recentRssis = AdvertisingPacketUtil.getRssisFromAdvertisingPackets(recentAdvertisingPackets);
-        return AdvertisingPacketUtil.getMeanRssi(recentRssis);
+    public float getDistance() {
+        RssiFilter filter = new ArmaFilter();
+        filter.setMaximumTimestamp(this.getLatestAdvertisingPacket().getTimestamp());
```

Comment:
No need for `this.`. Also you could just pass `maximumTimestamp` and `minimumTimestamp` in as construction parameter.
A convenience method for getting the latest timestamp may be useful too.